src/main/java/net/minecraftforge/fluids/FluidUtil.java

```diff
@@ -449,33 +492,100 @@ public static boolean tryPlaceFluid(@Nullable EntityPlayer player, World worldIn
         }
 
         // check that we can place the fluid at the destination
-        IBlockState destBlockState = worldIn.getBlockState(pos);
+        IBlockState destBlockState = world.getBlockState(pos);
         Material destMaterial = destBlockState.getMaterial();
         boolean isDestNonSolid = !destMaterial.isSolid();
-        boolean isDestReplaceable = destBlockState.getBlock().isReplaceable(worldIn, pos);
-        if (!worldIn.isAirBlock(pos) && !isDestNonSolid && !isDestReplaceable)
+        boolean isDestReplaceable = destBlockState.getBlock().isReplaceable(world, pos);
+        if (!world.isAirBlock(pos) && !isDestNonSolid && !isDestReplaceable)
         {
             return false; // Non-air, solid, unreplacable block. We can't put fluid here.
         }
 
-        if (worldIn.provider.doesWaterVaporize() && fluid.doesVaporize(fluidStack))
+        if (world.provider.doesWaterVaporize() && fluid.doesVaporize(fluidStack))
+        {
+            fluid.vaporize(player, world, pos, fluidStack);
+        }
+        else
+        {
+            if (!world.isRemote && (isDestNonSolid || isDestReplaceable) && !destMaterial.isLiquid())
+            {
+                world.destroyBlock(pos, true);
+            }
+
+            IFluidHandler handler = getPlacementFluidHandler(world, pos, fluid);
+            int consumed = handler.fill(fluidStack, false);
+            if (consumed == fluidStack.amount)
+            {
+                handler.fill(fluidStack, true);
+                SoundEvent soundevent = fluid.getEmptySound(fluidStack);
+                world.playSound(player, pos, soundevent, SoundCategory.BLOCKS, 1.0F, 1.0F);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Tries to place a fluid in the world in block form.
+     * Makes a fluid emptying sound when successful.
+     * Checks if water-like fluids should vaporize like in the nether.
+     *
+     * Modeled after {@link net.minecraft.item.ItemBucket#tryPlaceContainedLiquid(EntityPlayer, World, BlockPos)}
+     *
+     * @param player   Player who places the fluid. May be null for blocks like dispensers.
+     * @param world    World to place the fluid in
+     * @param resource The fluidStack to place.
+     * @param pos      The position in the world to place the fluid block
```

Comment:
Javadoc params do not match the function params here, please describe each param.